service/src/main/java/org/cbioportal/service/util/ProfiledCasesCounter.java

```diff
@@ -103,37 +106,37 @@ public void calculate(List<T> alterationCounts,
                 }
                 if (profiledCaseType == ProfiledCaseType.PATIENT) {
                     totalProfiledPatients.addAll(casesWithoutPanelData);
-                    alterationCountByGene.setNumberOfProfiledCases(totalProfiledPatients.size());
+                    alterationCount.setNumberOfProfiledCases(totalProfiledPatients.size());
                 } else {
                     totalProfiledSamples += casesWithoutPanelData.size();
-                    alterationCountByGene.setNumberOfProfiledCases(totalProfiledSamples);
+                    alterationCount.setNumberOfProfiledCases(totalProfiledSamples);
                 }
             } else {
                 // we use profiledCasesCount instead of casesWithoutPanelData to
                 // prevent a divide by zero error which can happen for targeted studies
                 // in which certain genes have events that are not captured by the panel.
-                alterationCountByGene.setNumberOfProfiledCases(profiledCasesCount);
+                alterationCount.setNumberOfProfiledCases(profiledCasesCount);
             }
-            alterationCountByGene.setMatchingGenePanelIds(allMatchingGenePanelIds);
+            alterationCount.setMatchingGenePanelIds(allMatchingGenePanelIds);
         }
 
         if (includeMissingAlterationsFromGenePanel) {
             Map<Integer, Boolean> genesWithAlteration = alterationCounts.stream()
-                    .collect(Collectors.toMap(AlterationCountByGene::getEntrezGeneId, x -> true));
+                    .flatMap(count -> Arrays.stream(count.getEntrezGeneIds()))
+                    .collect(Collectors.toMap(entrezGeneId -> entrezGeneId, x -> true));
```

Comment:
If we always set `true` as value, could we just use Set?

Reply:
Well spotted. I have changed this into a Set.